### PR TITLE
feat(generic-metrics): Implement use case awareness for Caching Indexer

### PR DESCRIPTION
**Pull request by john-z-yang**
### Stack PR Chain

| PR                                                   | Changes                                                  | Status        | Branch From                                          |
| ---------------------------------------------------- | -------------------------------------------------------- | ------------- | ---------------------------------------------------- |
| [#1](https://github.com/getsentry/sentry/pull/48309) | Implements use case awareness to RawSimpleIndexer        | Merged          | Master                                               |
| [#2](https://github.com/getsentry/sentry/pull/48350) | Implements use case awareness for Postgres Indexer       | Merged          | [#1](https://github.com/getsentry/sentry/pull/48309) |
| [#3](https://github.com/getsentry/sentry/pull/48475) | ➤ Implements use case awareness for Caching Indexer        | Merged | [#2](https://github.com/getsentry/sentry/pull/48350) |
| [#4](https://github.com/getsentry/sentry/pull/48355) | Implement use case awareness for Static String Indexer | Merged | [#3](https://github.com/getsentry/sentry/pull/48475) |
| [#5](https://github.com/getsentry/sentry/pull/49019) | Replace `bulk_record`/`record` with `_uca_bulk_record`/`_uca_record`      | Merged | [#4](https://github.com/getsentry/sentry/pull/48355) |

### Overview

See #47284

### CODE REVIEW

**Comment by ayirr7** [comment]
will fix tests next

File: src/sentry/sentry_metrics/indexer/cache.py [hunk 1]
```diff
@@ -38,63 +40,60 @@ def randomized_ttl(self) -> int:
         jitter = random.uniform(0, 0.25) * cache_ttl
         return int(cache_ttl + jitter)
 
-    def make_cache_key(self, key: str, cache_namespace: str) -> str:
-        hashed = md5_text(key).hexdigest()
-        return f"indexer:{self.partition_key}:org:str:{cache_namespace}:{hashed}"
+    def make_cache_key(self, key: str) -> str:
+        use_case_id, org_id, string = key.split(":", 2)
+        org_string = org_id + ":" + string
```
Comment (MeredithAnya) [inline src/sentry/sentry_metrics/indexer/cache.py:45]: do we need to split this up just to put it back together?

maybe we can just do 

```python
use_case_id, org_string = key.split(":", 1)
```

thoughts?

Comment (ayirr7) [reply src/sentry/sentry_metrics/indexer/cache.py:45]: that would also work I believe

File: tests/sentry/sentry_metrics/test_all_indexers.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def indexer_cls(request):
     return request.param
 
 
-@pytest.fixture(params=USE_CASE_KEYS)
+@pytest.fixture(params=UseCaseKey)
```
Comment (john-z-yang) [inline tests/sentry/sentry_metrics/test_all_indexers.py:47]: Nit: I'm happy with using the enum itself as an iterable, but we should get rid of the `USE_CASE_KEYS` in L27

**Comment by john-z-yang** [comment]
You probably need to also [fix test_indexer_cache.py](https://github.com/getsentry/sentry/blob/0929b22c591b1c5a6046c396dcbadc5b02afda9d/tests/sentry/sentry_metrics/test_indexer_cache.py) but I can help with that if you like

File: tests/sentry/sentry_metrics/test_indexer_cache.py [hunk 3]
```diff
@@ -17,39 +17,49 @@
...
+    key = indexer_cache.make_cache_key(f"{use_case_id}:{orgId}:{string}")
+
+    assert key == f"indexer:test:org:str:{use_case_id}:{nameSpace}"
 
 
 def test_formatted_results(use_case_id: str) -> None:
-    values = {"hello": 2, "bye": 3}
-    results = {indexer_cache.make_cache_key(k, use_case_id): v for k, v in values.items()}
-    assert indexer_cache._format_results(list(values.keys()), results, use_case_id) == values
+    values = {f"{use_case_id}::hello": 2, f"{use_case_id}::bye": 3}
```
Comment (MeredithAnya) [inline tests/sentry/sentry_metrics/test_indexer_cache.py:60]: why is there a double colon `::` here? is there supposed to be an org id in the string?

**Review by MeredithAnya — APPROVED** [review]
just had a smol question but I think overall LGTM

File: tests/sentry/sentry_metrics/test_indexer_cache.py (hunk 3 above)
Comment (john-z-yang) [reply tests/sentry/sentry_metrics/test_indexer_cache.py:60]: Just to see if the extra delimiter character will mess things up

**Pull request merged**